libpromises/evalfunction.c

```diff
@@ -7062,6 +7062,27 @@ static FnCallResult FnCallStringSplit(ARG_UNUSED EvalContext *ctx,
 
 /*********************************************************************/
 
+static FnCallResult FnCallStringReplace(ARG_UNUSED EvalContext *ctx,
+                                        ARG_UNUSED const Policy *policy,
+                                        ARG_UNUSED const FnCall *fp,
+                                        const Rlist *finalargs)
+{
+    char *string = RlistScalarValue(finalargs);
+    char *match = RlistScalarValue(finalargs->next);
+    char *substitute = RlistScalarValue(finalargs->next->next);
+
+    char *ret = SearchAndReplace(string, match, substitute);
```

Comment:
What does `SearchAndReplace` do if `match` is empty string? Personally, I think it should match nowhere and replace nothing.

Reply:
That's what it does, in contrast to regex_replace, which can replace the empty string.